packages/loader/driver-utils/src/readAndParse.ts

```diff
@@ -27,7 +28,6 @@ export async function readAndParse<T>(storage: Pick<IDocumentStorageService, "re
  * @returns the object that we decoded and JSON.parse
  */
 export function readAndParseFromBlobs<T>(blobs: {[index: string]: string}, id: string): T {
-    const encoded = blobs[id];
-    const decoded = fromBase64ToUtf8(encoded);
+    const decoded = blobs[id];
```

Comment:
I was talking about here. I know we will have follow up changes where blobs will be buffer. But now can I assume blobs[id] is already decoded? I looked at where this function readAndParseFromBlobs is called, which is just reading from different snapShotTrees.

Reply:
You should look around how readAndParseFromBlobs is used. 
I believe all usages are coming from ISnapshotTree.blobs, and ISnapshotTree is coming from driver (via ISnapshotTree.getSnapshotTree API).

To be honest, I'm a bit at a loss on what's in tree.blobs based on usage like this:
```
                    const channelAttributes = readAndParseFromBlobs<IChannelAttributes>(
                        tree.trees[path].blobs, tree.trees[path].blobs[".attributes"]);
```
It feels like ".attributes" resolves to an ID, which then used to look up ID in same blobs array and now it's base64 encoded content? Seems weird (or maybe I got confused).

Can you please debug through it to build better understanding of what's inside of blobs array?

Reply:
@jatgarg, can you please take look here? I looked a bit deeper, and I'm not sure understand or like what I see.
First, we use readAndParseFromBlobs() only for local paths, i.e. when either storage is not available, or we have local data store. I.e. it's only for drafts, as far as I understand.
The part that looks wrong is the fact that tree.blobs is used for two different purposes and has two different types of information:
1. A mapping from name to ID (i.e. ".attributes" -> some sha)
2. A mapping from ID to content (sha -> base64 encoded payload).

First, there might be collisions on names.
Second - we are changing everything to be buffers for data. The fact that data was string and happened to be of same type as hash type is an accident (both could have possibility to be not strings). We should not rely on this and allow any representation. The most important for this work, is that representation changes.

Can we use regular storage mechanism to reading these blobs, and build an adapter storage implementation that represents local content?

This will be blocking Si's progress...